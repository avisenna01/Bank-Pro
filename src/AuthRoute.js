import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import Cookies from 'universal-cookie';

export default function AuthRoute(ComponentToProtect) {
    return class extends Component {
        constructor() {
            super();
            this.state = {
                loading: true,
                redirect: false,
            };
        }
    }
}