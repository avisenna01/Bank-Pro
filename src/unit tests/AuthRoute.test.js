import React from 'react'
import AuthRoute from "../AuthRoute";
import renderer from 'react-test-renderer';
import ReactDOM from "react-dom";
import { shallow } from 'enzyme';

const rendercomponent = () => (shallow(<AuthRoute />));

describe('AuthRoute page try 1', () => {
    it('snapshot renders', () => {
        const renderedcomponent = rendercomponent();
        expect(renderedcomponent.find('div').length).toEqual(1);
    });
});

describe('AuthRoute page try 2', () => {
    test('snapshot renders', () => {
        const component = renderer.create(<AuthRoute />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AuthRoute />, div);
});