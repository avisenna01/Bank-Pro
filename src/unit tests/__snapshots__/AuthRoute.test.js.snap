// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthRoute page try 2 snapshot renders 1`] = `null`;
